backend/app/verification.py

```diff
@@ -135,8 +135,14 @@ def _get_domain_name(app_id: str) -> str | None:
         # https://sourceforge.net/p/forge/documentation/Project%20Web%20Services/
         return f"{projectname}.{domain}.io".lower()
     else:
-        [tld, domain] = app_id.split(".")[0:2]
-        domain = _demangle_name(domain)
+        compons = app_id.split(".")
+        compons.pop(-1) # remove the app name
+
+        domain = _demangle_name(compons.pop(-1))
```

Comment:
pop removes the element then returns that value, so domain is being set to the last component which is incorrect.

For something like `in.co.foo.bar` you are calculating the website to be `bar.foo.co.in` when it should be `foo.co.in`

Reply:
`return ".".join(reversed([ _demangle_name(i) for i in appid.split(".")[:-1]])).lower()`